src/test/java/io/pravega/connectors/flink/FlinkPravegaReaderTest.java

```diff
@@ -433,6 +499,30 @@ public boolean isEndOfStream(Integer nextElement) {
         }
     }
 
+    /**
+     * A deserialization schema with metadata for test purposes.
+     */
+    private static class TestDeserializationSchemaWithMetadata extends PravegaDeserializationSchema<Integer> {
+        public static final int END_OF_STREAM = -1;
+
+        public TestDeserializationSchemaWithMetadata() {
+            super(Integer.class, new IntegerSerializer());
+        }
+
+        @Override
+        public Integer deserializeWithMetadata(EventRead<Integer> eventRead) {
+            if (isEndOfStream(eventRead.getEvent())) {
+                return eventRead.getEvent();
+            }
+            return eventRead.getEventPointer().toBytes().remaining();
```

Comment:
My impression is that this method should return an object that contains the event data (an Integer in this case) as well as the EventPointer. This appears to be ignoring the event data and only returning the size of the EventPointer. How is this useful? I recommend changing this test so that it uses the metadata in a useful way. This should provide a good integration test as well as an example for how an end-user can actually use it.

Reply:
Also, use JSON for serialization as it is more realistic.

Reply:
Yes, when I wrote these, I was also trying to think in this unit test, whether I should keep it simple(hold the integer output just to test functionality) or keep it interpretable(more like a guiding example). I can change that with your suggestion.